### PR TITLE
add TryGetValue benchmarks, fixes #371

### LINKED ISSUE #371: Add new System.Collections.Immutable benchmark from recent CoreFX PR
`ImmutableDictionary.TryGetValue` was improved recently: https://github.com/dotnet/corefx/pull/35759.

We should consider adding benchmarks for it. An example benchmark is available in the PR itself.

cc @adamsitnik, @stephentoub

### PR DESCRIPTION
**Pull request by adamsitnik**
|                             Type |                    Method | Size |       Mean |     StdDev |
|--------------------------------- |-------------------------- |----- |-----------:|-----------:|
|   TryGetValueFalse<Int32, Int32> |                Dictionary |  512 |   7.377 us |  0.1738 us |
| TryGetValueFalse<String, String> |                Dictionary |  512 |  11.277 us |  0.0524 us |
|    TryGetValueTrue<Int32, Int32> |                Dictionary |  512 |   5.264 us |  0.0860 us |
|  TryGetValueTrue<String, String> |                Dictionary |  512 |  17.299 us |  1.2550 us |
|   TryGetValueFalse<Int32, Int32> |               IDictionary |  512 |   6.311 us |  0.3131 us |
| TryGetValueFalse<String, String> |               IDictionary |  512 |  12.309 us |  0.5233 us |
|    TryGetValueTrue<Int32, Int32> |               IDictionary |  512 |   7.351 us |  0.2442 us |
|  TryGetValueTrue<String, String> |               IDictionary |  512 |  16.916 us |  0.3923 us |
|   TryGetValueFalse<Int32, Int32> |                SortedList |  512 |  31.214 us |  0.7725 us |
| TryGetValueFalse<String, String> |                SortedList |  512 | 592.495 us | 18.5175 us |
|    TryGetValueTrue<Int32, Int32> |                SortedList |  512 |  31.001 us |  0.4571 us |
|  TryGetValueTrue<String, String> |                SortedList |  512 | 472.484 us | 30.5834 us |
|   TryGetValueFalse<Int32, Int32> |          SortedDictionary |  512 |  60.610 us |  0.4005 us |
| TryGetValueFalse<String, String> |          SortedDictionary |  512 | 706.629 us | 51.2005 us |
|    TryGetValueTrue<Int32, Int32> |          SortedDictionary |  512 |  55.691 us |  1.3485 us |
|  TryGetValueTrue<String, String> |          SortedDictionary |  512 | 526.923 us | 12.2692 us |
|   TryGetValueFalse<Int32, Int32> |      ConcurrentDictionary |  512 |   6.210 us |  0.2354 us |
| TryGetValueFalse<String, String> |      ConcurrentDictionary |  512 |  20.663 us |  0.6512 us |
|    TryGetValueTrue<Int32, Int32> |      ConcurrentDictionary |  512 |   6.174 us |  0.1445 us |
|  TryGetValueTrue<String, String> |      ConcurrentDictionary |  512 |  23.942 us |  1.3891 us |
|   TryGetValueFalse<Int32, Int32> |       ImmutableDictionary |  512 |  21.504 us |  0.2631 us |
| TryGetValueFalse<String, String> |       ImmutableDictionary |  512 |  46.049 us |  1.6524 us |
|    TryGetValueTrue<Int32, Int32> |       ImmutableDictionary |  512 |  24.911 us |  1.2669 us |
|  TryGetValueTrue<String, String> |       ImmutableDictionary |  512 |  56.956 us |  3.1367 us |
|   TryGetValueFalse<Int32, Int32> | ImmutableSortedDictionary |  512 |  35.945 us |  0.6413 us |
| TryGetValueFalse<String, String> | ImmutableSortedDictionary |  512 | 625.501 us | 46.8056 us |
|    TryGetValueTrue<Int32, Int32> | ImmutableSortedDictionary |  512 |  30.300 us |  0.3972 us |
|  TryGetValueTrue<String, String> | ImmutableSortedDictionary |  512 | 447.220 us |  7.9028 us |

### CODE REVIEW

File: src/benchmarks/micro/corefx/System.Collections/TryGetValue/TryGetValueTrue.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+            _immutableDictionary = Immutable.ImmutableDictionary.CreateRange<TKey, TValue>(_source);
+            _immutableSortedDictionary = Immutable.ImmutableSortedDictionary.CreateRange<TKey, TValue>(_source);
+        }
+
+        [Benchmark]
+        public bool Dictionary()
+        {
+            bool result = default;
+            var collection = _dictionary;
+            var found = _found;
```
Comment (stephentoub) [inline src/benchmarks/micro/corefx/System.Collections/TryGetValue/TryGetValueTrue.cs]: Nit: in most of this usage, var => whatever type it actually is

Comment (adamsitnik) [reply src/benchmarks/micro/corefx/System.Collections/TryGetValue/TryGetValueTrue.cs]: do you mean that we should use the exact types instead?

Comment (adamsitnik) [reply src/benchmarks/micro/corefx/System.Collections/TryGetValue/TryGetValueTrue.cs]: I have changed it to use the explicit types

File: src/benchmarks/micro/corefx/System.Collections/TryGetValue/TryGetValueFalse.cs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                result ^= collection.TryGetValue(notFound[i], out _);
+            return result;
+        }
+
+        [Benchmark]
+        [BenchmarkCategory(Categories.CoreCLR, Categories.Virtual)]
+        public bool IDictionary() => TryGetValue(_dictionary);
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public bool TryGetValue(IDictionary<TKey, TValue> collection)
```
Comment (ahsonkhan) [inline src/benchmarks/micro/corefx/System.Collections/TryGetValue/TryGetValueFalse.cs]: nit: private

File: src/benchmarks/micro/corefx/System.Collections/TryGetValue/TryGetValueTrue.cs (hunk 1 above)
Comment (stephentoub) [reply src/benchmarks/micro/corefx/System.Collections/TryGetValue/TryGetValueTrue.cs]: > do you mean that we should use the exact types instead

Yup. Thanks.

**Pull request merged**